tests/fuzz/s2n_certificate_extensions_parse_test.c

```diff
@@ -67,8 +64,13 @@ int s2n_fuzz_test(const uint8_t *buf, size_t len)
     POSIX_GUARD(s2n_stuffer_alloc(&fuzz_stuffer, len));
     POSIX_GUARD(s2n_stuffer_write_bytes(&fuzz_stuffer, buf, len));
 
+    DEFER_CLEANUP(struct s2n_config *config = s2n_config_new(), s2n_config_ptr_free);
+    EXPECT_NOT_NULL(config);
+    POSIX_GUARD(s2n_config_set_cipher_preferences(config, "20240503"));
```

Comment:
Yay for simplicity! The old function definitely seems a bit tricky.

It seems like there are two things the old function was doing, which were presumably important for some reason
1. enabling TLS 1.3
2. loading system certs

Can we call those out (+ why) in a comment here?

Reply:
> 1. enabling TLS 1.3

There is a [comment](https://github.com/aws/s2n-tls/blob/d467b80b941cc72ed66e95b3ebd4d0a952dc28b7/tests/fuzz/s2n_certificate_extensions_parse_test.c#L49) at the top of this file that we need tls13.


> 2. loading system certs

Ran locally and this [line](https://github.com/aws/s2n-tls/blob/535af43e66a24cdb3c847e9cab954ca5127975ce/tests/fuzz/s2n_certificate_extensions_parse_test.c#L99) fails if we remove cert loading so its implicitly documented.